ddtrace/appsec/processor.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os.path
+from typing import Any
+from typing import ClassVar
+from typing import Optional
+from typing import TYPE_CHECKING
+
+import attr
+
+import ddtrace
+from ddtrace import config
+from ddtrace.appsec._ddwaf import DDWaf
+from ddtrace.constants import MANUAL_KEEP_KEY
+from ddtrace.internal.logger import get_logger
+from ddtrace.utils.formats import get_env
+
+
+if TYPE_CHECKING:
+    from ddtrace import Span
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+DEFAULT_RULES = os.path.join(ROOT_DIR, "rules.json")
+
+log = get_logger(__name__)
+
+
+@attr.s(eq=False)
+class AppSecProcessor(object):
+
+    tracer = attr.ib(type=ddtrace.Tracer, default=None)
+    rules = attr.ib(type=str, default=None)
+    _ddwaf = attr.ib(type=DDWaf, default=None)
+
+    enabled = False
```

Comment:
Add typing; it's a ClassVar too from what I understand.
I'm not sure it's really needed. What about having:
```python
@property
def enabled(self) -> bool:
    return self._instance is not None
```

Reply:
I would like to keep accessing it on the class, maybe a `classproperty` would work but it's much more code.